moban/mobanfile/targets.py

```diff
@@ -34,6 +37,23 @@ def _handle_explicit_target(options, target):
     data_file = target.get(constants.LABEL_CONFIG, common_data_file)
     output = target[constants.LABEL_OUTPUT]
     template_type = target.get(constants.LABEL_TEMPLATE_TYPE)
+    if template_type and len(template_type) > 0:
+        if constants.TEMPLATE_TYPES_BASE_TYPE in template_type[0]:
+            adhoc_type = uuid.uuid4().hex
+            file_extension = uuid.uuid4().hex
+            base_type = template_type[0][constants.TEMPLATE_TYPES_BASE_TYPE]
+            template_types_options = template_type[1][
+                constants.TEMPLATE_TYPES_OPTIONS
+            ]
+            the_adhoc_type = {
+                adhoc_type: {
+                    "file_extensions": [file_extension],
```

Comment:
when the moban user mistakenly provide `file_extensions`, optionally we can note to the user that this field is not required for ad hoc template types.

Reply:
Updating the PR.